Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs

```diff
@@ -0,0 +1,317 @@
+
+                // Process pointer events as mouse events.
+                foreach(var p in pointerDataToUpdate)
+                {
+                    PointerData pointerData = p.Value;
+                    IMixedRealityPointer pointer = pointerData.pointer;
+
+                    if (pointer.IsInteractionEnabled
+                        && pointer.Rays != null
+                        && pointer.Rays.Length > 0
+                        && pointer.SceneQueryType == Physics.SceneQueryType.SimpleRaycast)
+                    {
+                        ProcessMouseEvent((int)pointer.PointerId);
+                    }
+                    else
+                    {
+                        // Remove Unity pointer if it exists, but leave the MRTK pointer in pointersData unless/until the pointer's source device is lost/disconnected.
+                        ProcessMrtkPointerLost(pointerData);
+                    }
+                }
+
+                for (int i = 0; i < pointerDataToRemove.Count; i++)
+                {
+                    ProcessMrtkPointerLost(pointerDataToRemove[i]);
+                }
+                pointerDataToRemove.Clear();
+            }
+            finally
+            {
+                Cursor.lockState = cursorLockStateBackup;
+            }
+
+            base.Process();
+        }
+
+        private void ProcessMrtkPointerLost(PointerData pointerData)
+        {
+            if (pointerData.lastMousePoint3d != null)
+            {
+                IMixedRealityPointer pointer = pointerData.pointer;
+
+                pointer.Result = null;
+                ProcessMouseEvent((int)pointer.PointerId);
+
+                pointerData.lastMousePoint3d = null; // Mark last mouse position invalid.
+            }
+        }
+
+        /// <summary>
+        /// Adds MRTK pointer support as mouse input for Unity UI.
+        /// </summary>
+        protected override MouseState GetMousePointerEventData(int id)
+        {
+            // Search for MRTK pointer with given id.
+            // If found, generate mouse event data for pointer, otherwise call base implementation.
+            PointerData pointerData;
+            if (pointerDataToUpdate.TryGetValue(id, out pointerData))
+            {
+                return GetMousePointerEventDataForMrtkPointer(pointerData);
+            }
+
+            return base.GetMousePointerEventData(id);
+        }
+
+        protected MouseState GetMousePointerEventDataForMrtkPointer(PointerData pointerData)
+        {
+            IMixedRealityPointer pointer = pointerData.pointer;
+
+            // Reset the RaycastCamera for projecting (used in calculating deltas)
+            Debug.Assert(pointer.Rays != null && pointer.Rays.Length > 0);
+            RaycastCamera.transform.position = pointer.Rays[0].Origin;
+            RaycastCamera.transform.rotation = Quaternion.LookRotation(pointer.Rays[0].Direction);
+
+            // Populate data
```

Comment:
>// Populate data [](start = 12, length = 16)

Nit: This comment doesn't seem to jive with what's done (i.e. "populate" vs "reset")

Reply:
Actually that comment refers to the whole section of setting pointerData,eventDataLeft. Not sure how to better communicate this?

Reply:
Ah got it - yeah it's tough without having some larger banner comment. This is fairly nit-y but in this case it might just be okay to delete that small comment - it doesn't really add that much to the code - it's clear from the code below that you're setting pointerData.eventDataLeft (i.e. you're populating data), and the rest of the code in this function already has good comments (explaining in detail what each population)

In the end, this is a super minor thing and up to you, I've just gotten feedback on comments before and have left comments similar to this in the past and reflecting on it have found that some were more helpful some were less, etc.